lib/awful/hotkeys_popup/widget.lua

```diff
@@ -37,6 +37,29 @@
 --
 --    hotkeys_popup.add_hotkeys(firefox_keys)
 --
+--
+-- Example of having different types of hotkey popups:
+--
+--    local globalkeys = awful.util.table.join(
+--        awful.key({modkey}, "/", function()
+--          hotkeys_popup.show_help()
+--        end, nil, {
+--          description = "show help (all)", group="HELP"
+--        }),
+--        awful.key({"Shift", modkey}, "/", function()
+--          hotkeys_popup.show_help(nil, nil, {show_awesome_keys=false})
+--        end, nil, {
+--          description = "show help for current app", group="HELP"
+--        }),
+--        awful.key({altkey, modkey}, "/", function()
+--          hotkeys_popup.show_help({}, nil, {show_awesome_keys=true})
+--        end, nil, {
+--          description = "show help for awesome only", group="HELP"
+--        })
+--        -- (more hotkeys goes here)
+--    )
```

Comment:
`awful.util.table` has been deprecated in favour of `gears.table`.
And the pattern `root.keys(globalkeys)` has been deprecated in favour of `awful.keyboard.append_global_keybindings(keys)` (where `keys` is a simple list, not `join`ed).

```suggestion
--    awful.keyboard.append_global_keybindings({
--        awful.key({modkey}, "/", function()
--          hotkeys_popup.show_help()
--        end, nil, {
--          description = "show help (all)", group="HELP"
--        }),
--        awful.key({"Shift", modkey}, "/", function()
--          hotkeys_popup.show_help(nil, nil, {show_awesome_keys=false})
--        end, nil, {
--          description = "show help for current app", group="HELP"
--        }),
--        awful.key({altkey, modkey}, "/", function()
--          hotkeys_popup.show_help({}, nil, {show_awesome_keys=true})
--        end, nil, {
--          description = "show help for awesome only", group="HELP"
--        })
--        -- (more hotkeys goes here)
--    })
```